core/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -712,6 +714,9 @@ private void parseSource(SearchContext context, SearchSourceBuilder source) thro
         if (source.minScore() != null) {
             context.minimumScore(source.minScore());
         }
+        if (source.profile() != null && source.profile()) {
```

Comment:
I am not far enough down the road but I don't get why we use the null invariant, why can't this simply return a primitive?